tinnsleep/create_reports.py

```diff
@@ -0,0 +1,149 @@
+            analyze
+        Returns
+        -------
+        picks_chan: array-like, shape (n_channels,)
+            list of str corresponding of the name of each channel to analyze
+        picks_imp:
+            list of str corresponding of the name of each channel of impedance values associated with the channels to
+            analyze
+        """
+    picks_imp = []
+    picks_chan = []
+    ch_names = raw.info["ch_names"]
+    for elm in ind_picks_imp:
+        picks_imp.append(ch_names[elm])
+    for elm in ind_picks_chan:
+        picks_chan.append(ch_names[elm])
+    return picks_chan, picks_imp
+
+
+def preprocess(raw, picks_chan, picks_imp, duration, interval, THR_imp=6000, get_log=False):
+    """Preprocesses raw for reporting
+    Parameters
+    ----------
+    raw: Instance of mne.Raw
+        the signal to analyze
+    picks_chan: array-like, shape (n_channels,)
+        list of str corresponding of the name of each channel to analyze
+    picks_imp:
+        list of str corresponding of the name of each channel of impedance values associated with the channels to
+        analyze
+    duration : int
+        Number of elements (i.e. samples) on the epoch.
+    interval: int
+        Number of elements (i.e. samples) to move for the next epoch.
+    THR_imp: float
+        Threshold value for the impedance rejection algorithm
+    get_log : OPTIONAL boolean default False
+        if True, create a report of the preprocessing rejecting steps
+    Returns
+    -------
+    epochs : ndarray, shape (n_epochs, n_channels, duration)
+        Epoched view of `data`. Epochs are in the first dimension.
+    valid_labels : lsit of booleans
+        labels of the epochs as good (True) or bad (False) for future annotation and reporting
+    log : dictionary
+        logs of the preprocessing steps, including the number of epochs rejected at each step
+
+    """
+    # Epoch rejection based on impedance
+    check_imp = Impedance_thresholding_sliding(raw[picks_imp][0], duration, interval, THR=THR_imp)
+    impedance_labels = np.any(check_imp, axis=-1)
+    suppressed_imp = np.sum(impedance_labels)
+
+    raw = CreateRaw(raw[picks_chan][0], picks_chan, ch_types=['emg'])  # pick channels and load
+    # Filtering data
+    raw = raw.filter(20., 99., n_jobs=4,
+                     fir_design='firwin', filter_length='auto', phase='zero-double',
+                     picks=picks_chan)
+
+    # Creating epochs
+    epochs = RawToEpochs_sliding(raw, duration=duration, interval=interval)
+
+    # Epoch rejection based on |min-max| thresholding
+    params = dict(ch_names=raw.info["ch_names"],
+                  rejection_thresholds=dict(emg=1e-04),  # two order of magnitude higher q0.01
+                  flat_thresholds=dict(emg=1e-09),  # one order of magnitude lower median
+                  channel_type_idx=dict(emg=[0, 1]),
+                  full_report=True
+                  )
```

Comment:
for your report you might want to have parameters not hardcoded. Specially `channel_type_idx` as this one can cause massive problems without you knowing !!!!!

Reply:
I placed params as a parameter of the preprocess function. It was definitely needed. Please adjust if needed. Please check notebook.

Reply:
validated !